docs/specs/saml.md

```diff
@@ -0,0 +1,705 @@
+
+- `<saml:Subject>`
+
+  - If provided, the subject will be passed to Authentication Flow for pre-selecting a user. See [Support Pre-Selecting User by SAML Subject in Authentication Flow](#1_5) for details.
+  - Ref: [SAMLCore](https://docs.oasis-open.org/security/saml/v2.0/saml-core-2.0-os.pdf) Section 3.3.4.
+
+- `<NameIDPolicy>`
+
+  - `<Format>`: If provided, `NameID` will be returned in the specified format. See [NameID](#3) section for supported formats.
+  - `<AllowCreate>`: If `true`, user can signup a new user during the SSO flow. Else, only login to an existing user is allowed. The default is `true`. Note that according to the SAML specification, the default value should be `false`, but we decided to change the default to `true` to match the behavior of the OIDC flow.
+  - `SPNameQualifier`: Not supported. If provided, it MUST be the entity ID of the service provider, or it will be an error.
+
+- `<saml:Conditions>`
+
+  - If provided, same conditions will be applied to the returned SAML assertions.
+
+- `<RequestedAuthnContext>`
+
+  - Not supported. If provided, only `<AuthnContextClassRef>` with value `urn:oasis:names:tc:SAML:2.0:ac:classes:unspecified` is accepted. Others will be an error.
+
+- `<Scoping>`
+
+  - Not supported. Will be ignored if provided.
+
+- `<ForceAuthn>`
+
+  - If `true`, any IdP Session will be ignored during the authentication process. i.e. The user must authenticate again by the Auth UI. If `<IsPassive>` is true, it is always an error.
+  - If `false`, and if an IdP Session already exist. Combining with `<IsPassive>`, there will be different behaviors:
+    - If `<IsPassive>` is `true`, the user will be authenticated directly without seeing any interactable UI.
+    - If `<IsPassive>` is `false`, the user will see the Select Account Screen. If `<saml:Subject>` was provided, the user will not be allowed to login with another account.
+  - Here are the equivalents of the `prompt` parameter in oidc:
+    - `prompt=login`: `ForceAuthn=true` and `IsPassive=false`.
+    - `prompt=select_account login`: `ForceAuthn=false` and `IsPassive=false`.
+    - `prompt=none`: `ForceAuthn=false` and `IsPassive=true`.
+
+- `<IsPassive>`
+
+  - See the above `<ForceAuthn>` section.
+
+- `<AssertionConsumerServiceIndex>`
+
+  - Not supported. Use `<AssertionConsumerServiceURL>` to specify the Assertion Consumer Service URL. Will be an error if provided.
+
+- `<AssertionConsumerServiceURL>`
+
+  - Optional. If not provided, the first item in the `acs_urls` config in items of `saml.service_providers` will be used.
+
+- `<ProtocolBinding>`
+
+  - Optional. If not provided, default to `urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Post`. See [Bindings](#5) section for supported bindings.
+
+- `<AttributeConsumingServiceIndex>`
+
+  - Not supported. Will be ignored if provided.
+
+- `<ProviderName>`
+  - Not supported. Will be ignored if provided.
+
+### <a id="1_5"></a> Support Pre-Selecting User by SAML Subject in Authentication Flow
```

Comment:
- [ ] Document new supported login_hint
- [ ] Document how subject is mapped to login_hint
- [ ] Document the behavior if IsPassive=true and Subject is given